encoding/protobuf/protoc-gen-yarpc-go/internal/testing/testing.pb.yarpc.go

```diff
@@ -237,23 +242,549 @@ func (h *_SinkYARPCHandler) Fire(ctx context.Context, requestMessage proto.Messa
 	if requestMessage != nil {
 		request, ok = requestMessage.(*FireRequest)
 		if !ok {
-			return protobuf.CastError(emptySink_FireYARPCRequest, requestMessage)
+			return protobuf.CastError(emptySinkServiceFireYARPCRequest, requestMessage)
+		}
+	}
+	return h.server.Fire(ctx, request)
+}
+
+func newSinkServiceFireYARPCRequest() proto.Message {
+	return &FireRequest{}
+}
+
+func newSinkServiceFireYARPCResponse() proto.Message {
+	return &yarpcproto.Oneway{}
+}
+
+var (
+	emptySinkServiceFireYARPCRequest  = &FireRequest{}
+	emptySinkServiceFireYARPCResponse = &yarpcproto.Oneway{}
+)
+
+// AllYARPCClient is the YARPC client-side interface for the All service.
+type AllYARPCClient interface {
+	GetValue(context.Context, *GetValueRequest, ...yarpc.CallOption) (*GetValueResponse, error)
+	SetValue(context.Context, *SetValueRequest, ...yarpc.CallOption) (*SetValueResponse, error)
+	Fire(context.Context, *FireRequest, ...yarpc.CallOption) (yarpc.Ack, error)
+	HelloOne(context.Context, ...yarpc.CallOption) (AllServiceHelloOneYARPCClient, error)
+	HelloTwo(context.Context, *HelloRequest, ...yarpc.CallOption) (AllServiceHelloTwoYARPCClient, error)
+	HelloThree(context.Context, ...yarpc.CallOption) (AllServiceHelloThreeYARPCClient, error)
+}
+
+// AllServiceHelloOneYARPCClient sends HelloRequests and receives the single HelloResponse when sending is done.
+type AllServiceHelloOneYARPCClient interface {
+	Context() context.Context
+	Request() *transport.StreamRequest
+	Response() *transport.StreamResponse
```

Comment:
We shouldn't expose transport-level objects in the encoding-level APIs.

Based on offline discussion: We need these for metadata access. We should use
`CallOption`s since we're already accepting them as part of the original call.